src/config_design.config

```diff
@@ -0,0 +1,126 @@
+%% Cloudant_benchmark config changes design review 
+
+%% ----------------------------------------------------------------------------
+%% CONFIGURATION TEMPLATES: (Fogbugz 62351)
+%% ----------------------------------------------------------------------------
+
+{config_templates, ["basho_bench", "cloudant_http", "http_driver"]}.
+
```

Comment:
I think for now we want to keep the template generation of configs outside of basho_bench, and focus on making basho_bench very good at running a single benchmark. Then we build wrapper functionality outside of basho_bench to handle generation of configs so that we can perform other actions like cluster setup/teardown, database setup/teardown, collection of additional metrics, etc. I think if we do the templating inside of basho_bench we're going to force ourselves to do everything else in there as well and I don't think that's the correct approach.